src/space_usage.rs

```diff
@@ -0,0 +1,123 @@
+}
+
+impl std::fmt::Display for SpaceUsage {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut report = String::from("SpaceUsage:\n");
+        let human_db_size = format_size(self.db_size, BINARY);
+        report += &format!("[Database Size]: {human_db_size}\n");
+        report += "[Largest Tables]:\n";
+        for (name, size, row_count) in &self.largest_tables {
+            let human_table_size = format_size(*size, BINARY);
+            report += &format!(
+                "   {name:<20} {human_table_size:>10}, {row_count:>6} rows\n",
+                name = format!("{name}:"),
+                row_count = row_count.map(|c| c.to_string()).unwrap_or("?".to_owned())
+            );
+        }
+        report += "[Webxdc With Biggest Status Update Space Usage]:\n";
+        for (msg_id, size, update_count) in &self.largest_webxdc_data {
+            let human_size = format_size(*size, BINARY);
+            report += &format!(
+                "   {msg_id:<8} {human_size:>10} across {update_count:>5} updates\n",
+                msg_id = format!("{msg_id}:")
+            );
+        }
+        write!(f, "{report}")
+    }
+}
+
+impl Context {
+    /// Get space usage information for the Context's database
+    /// used in Context.get_info()
+    pub async fn get_space_usage(&self) -> Result<SpaceUsage> {
+        // currently this is shown in system info, so needs to be fast,
+        // that's why we donot count size of all blobs for now
+        let page_size: usize = self
+            .sql
+            .query_get_value("PRAGMA page_size", ())
+            .await?
+            .unwrap_or_default();
+        let page_count: usize = self
+            .sql
+            .query_get_value("PRAGMA page_count", ())
+            .await?
+            .unwrap_or_default();
+
+        let mut largest_tables = Vec::new();
+
+        // check if https://sqlite.org/dbstat.html is enabled
+        if self
+            .sql
+            .query_map("SELECT * FROM dbstat LIMIT 1", (), |_| Ok(()), |_| Ok(()))
+            .await
+            .is_ok()
+        {
+            let biggest_tables = self
+                .sql
+                .query_map_vec(
+                    "SELECT name,
+                SUM(pgsize) AS size
+                FROM dbstat
+                WHERE name IN (SELECT name FROM sqlite_master WHERE type='table')
+                GROUP BY name ORDER BY size DESC LIMIT 10",
+                    (),
+                    |row| {
+                        let name: String = row.get(0)?;
+                        let size: usize = row.get(1)?;
+                        Ok((name, size))
+                    },
+                )
+                .await?;
+
+            for (name, size) in biggest_tables {
+                let row_count: Result<Option<usize>> = self
+                    .sql
+                    // SAFETY: the table name comes from the db, not from the user
```

Comment:
`SAFETY` is for `unsafe`: https://std-dev-guide.rust-lang.org/policy/safety-comments.html

I see we have a `SAFETY` comment already for `unwrap()`, but we should not add more, it's not for SQL injections or unwraps:
https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_safety_comment

Could be SECURITY or just a normal comment.